### DISCUSSION TITLE
Exporting micronet wind transducer readings via wifi or ESP-NOW?

### DISCUSSION THREAD

**Discussion by jeff-burright** [mnsnoop/esp32Micronet#7] [General]
Hi there, 
I have a tack tick wind instrument that came with the boat, and I'm trying to simply pull the wind speed and direction data into an ESP32 so I can shuttle those readings as simple numerical values to my ESP32 autopilot or a TTGO ESP32 smart watch. Not having dug too deep in this code yet, I see that it converts the incoming micronet data to NMEA2k, but I can't yet parse whether your code identifies the wind data as simple numerical variables that I could scrape and send via wifi or ESP-NOW to my other ESP32 devices. I might be doing this the hard way by not using NMEA2k, but it seems like a layer of abstraction more than I need and my boat system is pi-less. Can you please help me identify whether my desired use case is possible and where to find the variables I'm looking for? 
Thanks!

**Comment by mnsnoop** [comment]
My project doesn't actually use data from the micronet but the code is there for it to. Specifically lines 298 to 313 of micronet.cpp takes data from a micronet wind transducer and store it in the sDataArrayIn struct. There's no premade function to export that data so you'll have to code one or just set that struct as public. Micronet updates once a second so you can just poll that variable every second. There's also timestamps for each variable in that struct to track when it was last updated. 

80% of the project's code can be stripped away. Only things you need is the radio, micronet, and settings. I'd recommend keeping the printbuffer as it's great for debugging. 

I never tried running this with the wifi active. This project is basically bitbanging the Micronet protocol so needs very accurate timing and I'd assume wifi interrupts would step on it. You might be able to sleep the wifi right before the micronet transmission window. That'd be my first thought.

I no longer have Micronet gear so this project is concluded but I wish you luck on your!
